### PR TITLE
Feature/28470 author story cost

### PR DESCRIPTION
**Pull request by damonsharp**
- Add feature to plugin to allow author cost meta to be automatically associated with a post via ajax when that author is added via the Co-authors plus plugin.

### CODE REVIEW

File: js/global.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+( function( $ ) {
```
Comment (bcampeau) [inline js/global.js:1]: Weird tabbing in this file

File: post-payments.php [hunk 2]
```diff
@@ -22,6 +22,9 @@ public function __construct() {
 		add_action( 'admin_init', array( $this, 'enqueue' ) );
 		add_action( 'admin_menu', array( $this, 'add_tool_page' ) );
 		add_action( 'admin_init', array( $this, 'download_report' ) );
+        add_action( 'wp_ajax_link_author_post_cost_to_post', array( $this, 'link_author_post_cost_to_post' ) );
```
Comment (bcampeau) [inline post-payments.php]: Tabbing seems off here

File: post-payments.php [hunk 3]
```diff
@@ -46,6 +49,7 @@ public function add_meta_boxes() {
 			'attributes' => array( 'size' => 10, 'placeholder' => '0.00' ),
 		) );
 		$fm->add_meta_box( __( 'Story Cost', 'post-payments' ), $this->get_post_types(), 'normal', 'default' );
+        $fm->add_user_form( __( 'Story Cost', 'post-payments' ) );
```
Comment (bcampeau) [inline post-payments.php]: Tabbing seems off here

File: post-payments.php [hunk 4]
```diff
@@ -321,6 +324,48 @@ public function register_taxonomies() {
 		register_taxonomy( 'report-tags', array( 'post' ), $args );
 	}
 
+    /**
+     * Ajax callback for linking author post cost to post
+     *
+     * @return void
+     */
+    public function link_author_post_cost_to_post() {
```
Comment (bcampeau) [inline post-payments.php]: If this function is used on the frontend, the output should be cached

File: post-payments.php [hunk 5]
```diff
@@ -321,6 +324,48 @@ public function register_taxonomies() {
...
+                        echo json_encode( $user_post_cost );
+                    }
+                }
+            }
+        }
+        exit;
+    }
+
+    public function admin_enqueue_scripts() {
+        global $post;
```
Comment (bcampeau) [inline post-payments.php]: Why is this here? Doesn't seem to be used in this function

**Comment by bcampeau** [comment]
Please address my feedback. Thanks!

**Comment by damonsharp** [comment]
$post definitely not needed. Sublime is adding in spaces instead of tabs sometimes. Looking into that, but should be fixed here. On caching, this functionality is only on the backend when editing a post.

**Comment by bcampeau** [comment]
:horse:
